pkg/avancement/service_manager_test.go

```diff
@@ -254,3 +255,28 @@ func (s *mockSource) AddFiles(name string) {
 	}
 	s.files = append(s.files, path.Join(s.localPath, name))
 }
+
+func TestRepositoryCloneErrorOmitsToken(t *testing.T) {
+	dstBranch := "test-branch"
+	author := &git.Author{Name: "Testing User", Email: "testing@example.com", Token: "test-token"}
+	client, _ := fakescm.NewDefault()
+	fakeClientFactory := func(s string) *scm.Client {
+		return client
+	}
+	sm := New("tmp", author)
+	sm.clientFactory = fakeClientFactory
+
+	sm.repoFactory = func(url, _ string, _ bool) (git.Repo, error) {
+		// This actually causes the error and results in trying to create a repository
+		// which can surface the token
+		errorMessage := fmt.Errorf("failed to clone repository %s: exit status 128", dev)
+		return nil, errorMessage
+	}
+	err := sm.Promote("my-service", dev, staging, dstBranch, "foo", false)
+	if err != nil {
+		// Useful debug to check all is working OK and what the message is:
```

Comment:
Should these comments and the t.Fatalf still be here?

Reply:
I'll remove now, it's pretty trivial to figure out anyway